src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanNoArgConstructorQuickFix.java

```diff
@@ -37,57 +44,66 @@
  *
  */
 
-public class ManagedBeanNoArgConstructorQuickFix {
+public class ManagedBeanNoArgConstructorQuickFix implements IJavaCodeActionParticipant {
+    private static final Logger LOGGER = Logger.getLogger(ManagedBeanNoArgConstructorQuickFix.class.getName());
+
+    @Override
+    public String getParticipantId() {
+        return ManagedBeanNoArgConstructorQuickFix.class.getName();
+    }
 
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
-        PsiElement node = context.getCoveredNode();
-        PsiClass parentType = getBinding(node);
+        return addConstructor(diagnostic, context);
+    }
+
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
+        final PsiElement node = context.getCoveredNode();
+        final PsiClass parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
+
         if (parentType != null) {
```

Comment:
It seems this check was being made before in getCodeActions() to determine whether a code action should even be created and has been moved into resolveCodeAction(). This seems to be too late. I think this might produce a quick fix that does nothing when the user clicks on it, whereas before no quick fix would be presented to the user.